report/thesis.tex

```diff
@@ -75,25 +92,34 @@ \chapter{Tendermint protocol overview and the fork accountability problem}
+As we can see in Figure 2.1, the consensus algorithm proceeds in rounds: in each round there is a validator (i.e., a process participating in the consensus) that proposes a value (proposer) and all the validators then vote during the round on whether to accept the proposed value or move on to the next round.
 
-During a round, processes can exchange different types of messages:
+We can summarize the execution in a round in **two voting steps**: PREVOTE and PRECOMMIT steps. A vote can be for a particular value or for *nil* (null value). During a round, processes can exchange different types of messages, each one with a specific purpose:
  
 - **PROPOSAL message**: sent by the proposer of the round to propose a value in a new round
 
 - **PREVOTE message**: sent by validators to vote for a value during the first voting step
 
 - **PRECOMMIT message**: sent by validators to vote for a value during the second voting step
- 
-We can summarize the execution in a round in **two voting steps**: PREVOTE and PRECOMMIT steps. A vote can be for a particular value or for *nil* (null value).
 
-A correct process decides on a value *v* in a round *r* upon receiving a proposal and *2f + 1* quorum of PRECOMMIT messages for *v* in a round *r*. 
-A correct process sends a PRECOMMIT message for a value *v* in a round *r* upon receiving a proposal and *2f + 1* quorum of PREVOTE messages for *v* in a round *r*.
+A correct process decides on a value *v* in a round *r* upon receiving a proposal and quorum of at least *2f + 1* PRECOMMIT messages for *v* in a round *r*. 
+A correct process sends a PRECOMMIT message for a value *v* in a round *r* upon receiving a proposal and a quorum of at least *2f + 1* PREVOTE messages for *v* in a round *r*.
+
+Validators wait some time before sending a PREVOTE for *nil* if they do not receive a valid proposal after a certain time (*proposalTimeout*) and they send a PRECOMMIT message for *nil* if they do not receive at least *2f + 1* matching PREVOTE messages for a value after a certain time (*prevoteTimeout*).
+If processes do not receive at least *2f + 1* matching PRECOMMIT messages for a value after a certain time (*precommitTimeout*), they move to the next round.
+
+After a decision has been made, processes continue to agree on other values on another consensus instance (*height*) and they repeat the process described above in order to agree on different blocks.
+
+\end{markdown}
 
-Validators wait some time before sending a PREVOTE for *nil* if they do not receive a valid proposal after a certain time (*proposalTimeout*) and they send a PRECOMMIT message for *nil* if they do not receive *2f + 1* matching PREVOTE messages for a value after a certain time (*prevoteTimeout*).
-If processes do not receive *2f + 1* matching PRECOMMIT messages for a value after a certain time (*precommitTimeout*), they move to the next round.
+\begin{figure}[h]
+\centering
+\includegraphics[scale=0.45]{tendermint_schema.png} 
```

Comment:
This figure looks nice, it could go in your presentation. I would avoid going into too much details, however, about the consensus algorithm, since there are many subtleties that can be tricky to explain. So I would stay high level.

Reply:
Agree, will do.